### PR TITLE
UCAI core package metadata update and fix

### PR DESCRIPTION
**Pull request by serena-ruan**
**PR Checklist**

- [ ] A description of the changes is added to the description of this PR.
- [ ] If there is a related issue, make sure it is linked to this PR.
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added or modified a feature, documentation in `docs` is updated

**Description of changes**
Package metadata update and fix README

### CODE REVIEW

**Comment by serena-ruan** [comment]
@BenWilson2 Could you take a look at this one?

File: unitycatalog-ai/pyproject.toml [hunk 1]
```diff
@@ -13,6 +13,36 @@ dependencies = [
...
+  "Intended Audience :: Education",
+  "Intended Audience :: End Users/Desktop",
+  "Intended Audience :: Science/Research",
+  "Intended Audience :: Information Technology",
+  "Intended Audience :: Other Audience",
+  "License :: OSI Approved :: Apache Software License",
+  "Operating System :: OS Independent",
+  "Programming Language :: Python :: 3",
+  "Programming Language :: Python :: 3.9",
+  "Programming Language :: Python :: 3.10",
```
Comment (BenWilson2) [inline unitycatalog-ai/pyproject.toml:29]: Let's verify that things work on 3.11 / 3.12, update if it does.

**Review by BenWilson2 — APPROVED** [review]
LGTM once we validate py support!

File: unitycatalog-ai/pyproject.toml (hunk 1 above)
Comment (BenWilson2) [reply unitycatalog-ai/pyproject.toml:29]: Awesome!

**Comment by BenWilson2** [comment]
@dennyglee we're all set on this one! (We'll be filing a follow-up for namespace changes to conform to `unitycatalog-x` instead of `ucai-x` (hopefully today) :)

**Review by dennyglee — APPROVED** [review]
Got it, thanks @serena-ruan and @BenWilson2

**Pull request merged**